model/GameModel.ts

```diff
@@ -1,16 +1,97 @@
+import {resolvePromise} from "./resolvePromise.js";
+
 export class GameModel {
+
+    //celebrity information
     private _name: string;
-    private _imageUrl: string;
-    private _birthYear: number;
-    private _blur: number = 0;
+    private _imageUrl: string = '';
+    private _birth: Hint<Date> = new Hint(1, new Date(0,0,0), true);
+    private _death : Hint<Date> = new Hint(1, new Date(0,0,0));
+    private _occupation : Hint<string>= new Hint(1, '');
+    private _citizenship : Hint<string> = new Hint(1, '');
+    private _initials : Hint<string> = new Hint(2, '');
+    private _paragraph1 : Hint<string> = new Hint (2, '');
+
+    //game information
+    private _blur: number = 4;
+    private _curHintLevel: number = 1;
+    private _hints: Hint<any>[] = [this._death, this._occupation, this._citizenship, this._initials, this._paragraph1];
+    private _nbGuesses = 0;
+    private _curGuess : string = '';
+    private _prevGuesses : string[] = [];
+    private _promiseState: any = {};
+    private _end: boolean = false;
+    private _win: boolean = false;
 
     /**
      * Model for the game
      */
-    constructor(name: string, imageUrl: string, birthYear: number) {
+    constructor(name: string) {
         this._name = name;
-        this._imageUrl = imageUrl;
-        this._birthYear = birthYear;
+        resolvePromise(this.dummyPromise(), this._promiseState); //TODO: change the promise to a call to the api
```

Comment:
In the constructor, we could call a method which would initialize the hints list instead of initializing them one-by-one directly in the constructor. This auxiliary method would use the parsing methods provided by the api.

Reply:
es sure, but we can't change that now since we don't have parsing methods implemented, but yes I wrote a TODO that this should be changed once parsing is implemented